app/experimenter/nimbus-ui/src/hooks/useCommonForm.tsx

```diff
@@ -7,26 +7,51 @@ import { RegisterOptions, FieldError } from "react-hook-form";
 import Form from "react-bootstrap/Form";
 import { useForm } from "react-hook-form";
 import { camelToSnakeCase } from "../lib/caseConversions";
+import { branchFieldNames } from "../components/FormBranches";
 
 // TODO: 'any' type on `onChange={(selectedOptions) => ...`,
 // it wants this, but can't seem to coerce it into SelectOption type
 // type SelectedOption = {
 //   value: ValueType<any, true>;
 //   action: ActionMeta<any>;
 // };
+
 export type SelectOption = { label: string; value: string };
+export interface UseCommonNestedForm {
+  (
+    prefix: string,
+    nestedSubmitErrors: Record<string, string[]>,
+    nestedErrors: Record<string, FieldError>,
+    nestedTouched: Record<string, boolean>,
+  ): {
+    FormErrors: ({
+      name,
+    }: {
+      name: typeof branchFieldNames[number];
```

Comment:
@lmorchard Something is not quite right here and could use a hand with your TS foo. 🤨 You can see the full error hovering over `useCommonNestedForm` - I've tried a few different variations of the signature as well as what `K` extends in the hook and am missing some piece of the puzzle...

Reply:
Can you paste the error you're seeing? I checked out this branch but I'm not seeing an error here on hover. I do see this error a few lines down at 35 when I run `yarn lint`, though?
```
/home/lmorc/devel/experimenter/app/experimenter/nimbus-ui/src/hooks/useCommonForm.tsx
  35:10  error  Don't use 'Object' as a type. Use Record<string, any> instead  @typescript-eslint/ban-types
```

Reply:
L166 should be showing this big nasty:

```
Type '(prefix: string, nestedSubmitErrors: Record<string, string[]>, nestedErrors: Record<string, FieldError>, nestedTouched: Record<string, boolean>) => { ...; }' is not assignable to type 'UseCommonNestedForm'.
  Call signature return types '{ FormErrors: <K extends FieldNames>({ name }: { name: K; }) => Element; formControlAttrs: <K extends FieldNames>(name: K, registerOptions: Partial<{ required: string | boolean | ValidationValueMessage<boolean>; ... 8 more ...; setValueAs: (value: any) => any; }>) => { ...; }; watch: { ...; }; }' and '{ FormErrors: ({ name, }: { name: "name" | "description" | "ratio" | "featureValue" | "featureEnabled"; }) => Element; formControlAttrs: (name: "name" | "description" | "ratio" | "featureValue" | "featureEnabled", registerOptions?: Partial<...> | undefined) => Object; watch: any; }' are incompatible.
    The types of 'FormErrors' are incompatible between these types.
      Type '<K extends FieldNames>({ name }: { name: K; }) => JSX.Element' is not assignable to type '({ name, }: { name: "name" | "description" | "ratio" | "featureValue" | "featureEnabled"; }) => Element'.
        Types of parameters '__0' and '__0' are incompatible.
          Type '{ name: "name" | "description" | "ratio" | "featureValue" | "featureEnabled"; }' is not assignable to type '{ name: FieldNames; }'.
            Types of property 'name' are incompatible.
              Type 'string' is not assignable to type 'FieldNames'.
                'string' is assignable to the constraint of type 'FieldNames', but 'FieldNames' could be instantiated with a different subtype of constraint 'string'.
                  Type 'string' is not assignable to type 'FieldNames'.
                    'string' is assignable to the constraint of type 'FieldNames', but 'FieldNames' could be instantiated with a different subtype of constraint 'string'.ts(2322)
```